libs/ledger-live-common/src/families/avalanchepchain/js-signOperation.ts

```diff
@@ -0,0 +1,366 @@
+          (e) => o.error(e)
+        );
+        return () => {
+          cancelled = true;
+        };
+      })
+  );
+
+const signTransactionParsable = async <
+  UnsignedTx extends PlatformUnsignedTx,
+  SignedTx extends PlatformTx
+>(
+  unsignedTx: UnsignedTx,
+  paths: string[],
+  avalanche
+): Promise<SignedTx> => {
+  const accountPath = BIPPath.fromString(AVAX_BIP32_PREFIX);
+  const bip32Paths = pathsToUniqueBipPaths(paths);
+  const txbuff = unsignedTx.toBuffer();
+  const changePath = BIPPath.fromString(`${AVAX_BIP32_PREFIX}/0/0`);
+
+  const ledgerSignedTx = await avalanche.signTransaction(
+    accountPath,
+    bip32Paths,
+    txbuff,
+    changePath
+  );
+
+  const sigMap = ledgerSignedTx.signatures;
+  const credentials = getCredentials<UnsignedTx>(unsignedTx, paths, sigMap);
+  const signedTx = new PlatformTx(
+    unsignedTx as PlatformUnsignedTx,
+    credentials
+  );
+
+  return signedTx as SignedTx;
+};
+
+const signTransactionHash = async <
+  UnsignedTx extends PlatformUnsignedTx,
+  SignedTx extends PlatformTx
+>(
+  unsignedTx: UnsignedTx,
+  paths: string[],
+  avalanche
+): Promise<SignedTx> => {
+  const txbuff = unsignedTx.toBuffer();
+  const msg = Buffer.from(createHash("sha256").update(txbuff).digest());
+  const bip32Paths = pathsToUniqueBipPaths(paths);
+  const accountPath = BIPPath.fromString(AVAX_BIP32_PREFIX);
+  const sigMap = await avalanche.signHash(accountPath, bip32Paths, msg);
+  const creds: Credential[] = getCredentials<UnsignedTx>(
+    unsignedTx,
+    paths,
+    sigMap
+  );
+  const signedTx = new PlatformTx(unsignedTx as PlatformUnsignedTx, creds);
+
+  return signedTx as SignedTx;
+};
+
+const getCanLedgerParse = (config, unsignedTx) => {
+  let canLedgerParse = config.version >= "0.3.1";
+
+  const txIns = unsignedTx.getTransaction().getIns();
+
+  for (let i = 0; i < txIns.length; i++) {
+    const typeID = txIns[i].getInput().getTypeID();
+    if (typeID === STAKEABLELOCKINID) {
+      canLedgerParse = false;
+      break;
+    }
+  }
+
+  return canLedgerParse;
+};
+
+const pathsToUniqueBipPaths = (paths: string[]) => {
+  const uniquePaths = paths.filter((val: any, i: number) => {
+    return paths.indexOf(val) === i;
+  });
+
+  const bip32Paths = uniquePaths.map((path) => {
+    return BIPPath.fromString(path, false);
+  });
+
+  return bip32Paths;
+};
+
+const getTransactionPathsAndAddresses = (unsignedTx, chainId, pAddresses) => {
+  unsignedTx.toBuffer();
+  const tx = unsignedTx.getTransaction();
+
+  const items = tx.getIns();
+  let operations: TransferableOperation[] = [];
+
+  // Try to get operations, it will fail if there are none, ignore and continue
+  try {
+    operations = (tx as OperationTx).getOperations();
+    // eslint-disable-next-line no-empty
+  } catch (e) {}
+
+  const hrp = AVAX_HRP;
+  const paths: string[] = [];
+  const addresses: string[] = [];
+
+  // Collect derivation paths for source addresses
+  for (let i = 0; i < items.length; i++) {
```

Comment:
We use the index later in the file to calculate `pathIndex`. Since we get easier access to the index in a `for` loop than a `for...of` loop, would you be okay keeping these as `for` loops?

Reply:
If you need index to do i + j, you can keep those loops but if you don't i would suggest to go with a `for...of...` or even a forEach()

Small tips if you want to have index in a `for ... of ...` loop
(personally I don't like it that much, in that case I would better use classical `for` loop)
```javascript
for (const [i, v] of [1, 2, 3, 4, 5].entries()) {
  console.log(i, v);
//0,1
//1,2
....
//4,5
}
```